### ISSUE TITLE
Disallow space before colon in SwitchStatement

### ISSUE THREAD

**Issue by kohver** [eslint/eslint#7981]
Hi guys!
Today there is no way to disallow space before colon in switch-case statement:
```js
switch (foo) {
  case 'bar' : 
    return '^---oh, that space...';
}
```
I already use `keyword-spacing`, but I can't configure it for my case. I tried `key-spacing` too, but it still doesn't work.
My configuration looks like this:
```
    ...
    "key-spacing": "error",
    "keyword-spacing": "error",
    ...
```
You can also reproduce this by http://eslint.org/demo/ with enabled `keyword-spacing` & `key-spacing`.

I've asked about this in gitter and I was recommended to create an issue. I don't know, should it be a new rule or modification of an existing?

Thank you!

**Comment by nzakas** [comment]
@eslint/eslint-team looks like we lost track of this. Anyone want to champion this proposal?

**Comment by mikesherov** [comment]
@nzakas I'm willing to champion this one. However, this seems like it doesn't belong with keyword-spacing nor key-spacing. Do we yet have a rule for switch spacing? If not, should we create one? Also, I suppose it should handle a space after the colon (in the case of a single line case)?

**Comment by mysticatea** [comment]
I'll champion, too.
I try to make a concrete proposal.

## label-colon-spacing (or label-spacing)

This rule enforce usage of spaces before/after the colon of labels. The labels is both `LabeledStatement` and `SwitchCase` nodes'.

```js
A_LABEL : while (true) foo();
//     ^ ^

switch (a) {
    case A : foo(); break;
    //    ^ ^
    case B :
    //    ^ (if there is a line terminator after the colon, this rule does nothing for spacing after the colon.)
    default : foo(); break;
    //     ^ ^
}
```

### Options

```json
{
    "label-spacing": ["error", {"after": true, "before": false}]
}
```

- `after`
    - `true` (default) requires spaces after the colon of labels.
    - `false` disallows spaces after the colon of labels.
- `before`
    - `true` requires spaces before the colon of labels.
    - `false` (default) disallows spaces before the colon of labels.

### Examples

:-1: incorrect code for this rule with the default `{"after": true, "before": false}` option.

```js
A_LABEL :foo();

switch (a) {
    case A :foo();
    default :foo();
}
```

:+1: correct code for this rule with the default `{"after": true, "before": false}` option.

```js
A_LABEL: foo();

switch (a) {
    case A: foo();
    default: foo();
}
```

:-1: incorrect code for this rule with the `{"after": false, "before": true}` option.

```js
A_LABEL: foo();

switch (a) {
    case A: foo();
    default: foo();
}
```

:+1: correct code for this rule with the default `{"after": false, "before": true}` option.

```js
A_LABEL :foo();

switch (a) {
    case A :foo();
    default :foo();
}
```

**Comment by mikesherov** [comment]
I think we need two separate rules for labels and switch cases, as they are really two separate things and I imagine folks will be looking for "switch case spacing" as a search term nor then "label spacing". 

Thoughts?

**Comment by nzakas** [comment]
Agree, I don't think it makes sense to group labels and switch cases. I'd argue just going with switch cases for now and see if anyone requests one for labels in the future.

**Comment by not-an-aardvark** [comment]
@eslint/eslint-team We need one more :+1: from team members to accept this issue -- anyone willing to support it?

**Comment by kaicataldo** [comment]
I support adding a rule for this, however it's not entirely clear to me where we stand with the proposal. It looks like there were two votes for a rule that specifically checks switch cases.
